### PR TITLE
[core] Support creating tag

### LINKED ISSUE #1270: [Feature] Support to create tag
### Search before asking

- [X] I searched in the [issues](https://github.com/apache/incubator-paimon/issues) and found nothing similar.

### Motivation

Sub task of #1269.

### Solution

_No response_

### Anything else?

_No response_

### Are you willing to submit a PR?

- [X] I'm willing to submit a PR!

### PR DESCRIPTION
**Pull request by yuzelin**
### Purpose

close #1270 

### Tests

`FileStoreTableTestBase#testCreateTag`

### CODE REVIEW

File: paimon-core/src/main/java/org/apache/paimon/AbstractFileStore.java [hunk 1]
```diff
@@ -181,4 +184,15 @@ public PartitionExpire newPartitionExpire(String commitUser) {
                 newScan(),
                 newCommit(commitUser));
     }
+
+    @Override
+    public void createTag(String tagName) {
```
Comment (JingsongLi) [inline paimon-core/src/main/java/org/apache/paimon/AbstractFileStore.java:189]: Just in table level, don't need in store level.

File: paimon-core/src/main/java/org/apache/paimon/table/FileStoreTable.java [hunk 2]
```diff
@@ -90,4 +90,6 @@ default Optional<String> comment() {
     default BinaryTableStats getSchemaFieldStats(DataFileMeta dataFileMeta) {
         return dataFileMeta.valueStats();
     }
+
+    void createTag(String tagName);
```
Comment (JingsongLi) [inline paimon-core/src/main/java/org/apache/paimon/table/FileStoreTable.java:94]: create this method in `Table`

File: paimon-core/src/main/java/org/apache/paimon/utils/TagManager.java [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+        } catch (IOException e) {
+            throw new RuntimeException(
+                    String.format(
+                            "Failed to determine if tag '%s' exists in path %s.", tagName, path),
+                    e);
+        }
+    }
+
+    /** Get the tagged snapshot. */
+    public Snapshot snapshot(String tagName) {
```
Comment (JingsongLi) [inline paimon-core/src/main/java/org/apache/paimon/utils/TagManager.java:76]: tagSnapshot

File: paimon-core/src/main/java/org/apache/paimon/table/FileStoreTable.java (hunk 2 above)
Comment (JingsongLi) [inline paimon-core/src/main/java/org/apache/paimon/table/FileStoreTable.java:94]: `createTag(String name, long snapshotId)` is better?

File: paimon-core/src/main/java/org/apache/paimon/table/AbstractFileStoreTable.java [hunk 4]
```diff
@@ -243,4 +246,19 @@ public void rollbackTo(long snapshotId) {
             throw new UncheckedIOException(e);
         }
     }
+
+    @Override
+    public void createTag(String tagName, long fromSnapshotId) {
+        checkArgument(!StringUtils.isNumeric(tagName), "Tag name cannot be pure numeric string.");
```
Comment (JingsongLi) [inline paimon-core/src/main/java/org/apache/paimon/table/AbstractFileStoreTable.java:252]: inline `isNumeric` here.
And add test for empty "" tag string.
Move this check to `TagManager`.

File: paimon-core/src/main/java/org/apache/paimon/table/Table.java [hunk 5]
```diff
@@ -67,6 +67,9 @@ public interface Table extends Serializable {
     @Experimental
     void rollbackTo(long snapshotId);
 
+    /** Create a tag from given snapshot. */
```
Comment (JingsongLi) [inline paimon-core/src/main/java/org/apache/paimon/table/Table.java:70]: add `@Experimental`

**Review by JingsongLi — APPROVED** [review]
+1
